SemiconductorTestLibrary.Extensions/source/InstrumentAbstraction/DCPower/Source.cs

```diff
@@ -222,7 +298,121 @@ public static void ForceCurrent(this DCPowerSessionsBundle sessionsBundle, doubl
         }
 
         /// <summary>
-        /// Forces current and specifies symmetric voltage limits.
+        /// Forces current on the target pins at the specified level. Must at least provide a level value, and the method will assume all other properties that have been previously set. Optionally, can also provide a specific voltage limit, current level range, voltage limit range values directly.
+        /// </summary>
+        /// <param name="sessionsBundle">The <see cref="DCPowerSessionsBundle"/> object.</param>
+        /// <param name="currentLevels">The current level to force for different pins.</param>
+        /// <param name="voltageLimit">The voltage limit to use.</param>
+        /// <param name="currentLevelRange">The current level range to use.</param>
+        /// <param name="voltageLimitRange">The voltage limit range to use.</param>
+        /// <param name="waitForSourceCompletion">Setting this to True will wait until sourcing is complete before continuing, which includes the set amount of source delay.
+        /// Otherwise, the source delay amount is not directly accounted for by this method and the WaitForEvent must be manually invoked in proceeding code.</param>
+        public static void ForceCurrent(this DCPowerSessionsBundle sessionsBundle, IDictionary<string, double> currentLevels, double? voltageLimit = null, double? currentLevelRange = null, double? voltageLimitRange = null, bool waitForSourceCompletion = false)
+        {
+            sessionsBundle.Do((sessionInfo, sitePinInfo) =>
+            {
+                var settings = new DCPowerSourceSettings()
+                {
+                    OutputFunction = DCPowerSourceOutputFunction.DCCurrent,
+                    LimitSymmetry = DCPowerComplianceLimitSymmetry.Symmetric,
+                    Level = currentLevels[sitePinInfo.PinName],
+                    Limit = voltageLimit,
+                    LevelRange = currentLevelRange,
+                    LimitRange = voltageLimitRange
+                };
+                sessionInfo.Force(settings, sitePinInfo.IndividualChannelString, waitForSourceCompletion: waitForSourceCompletion);
+            });
+        }
+
+        /// <summary>
+        /// Forces current on the target pins at the specified level. Must at least provide a level value, and the method will assume all other properties that have been previously set. Optionally, can also provide a specific voltage limit, current level range, voltage limit range values directly.
```

Comment:
Suggest changing to: "Forces current on the target pins at the specified site-unique level."